management/src/clusterm/manager/api.go

```diff
@@ -141,36 +137,35 @@ func validateAndSanitizeEmptyExtraVars(errorPrefix, extraVars string) (string, e
 	return extraVars, nil
 }
 
-func (m *Manager) nodesCommission(tags []string, sanitizedExtraVars string) error {
-	me := newWaitableEvent(newCommissionEvent(m, tags, sanitizedExtraVars))
+func (m *Manager) nodesCommission(req *APIRequest, extraVars string) error {
+	if !IsValidHostGroup(req.HostGroup) {
```

Comment:
But isn't even that a part of Job Processing? I think we should not even be posting a job is some mandatory params are missing. Fail early.  Why go and setup things in collins/boltdb